packages/openapi-v3/src/__tests__/integration/controller-spec.integration.ts

```diff
@@ -307,6 +307,71 @@ describe('controller spec', () => {
       });
     });
 
+    it('allows operations to get definitions of models when defined through a different method', async () => {
+      @model()
+      class Todo extends Entity {
+        @property({
+          type: 'string',
+          required: true,
+        })
+        title: string;
+      }
+
+      class MyController {
+        @get('/todos', {
+          responses: {
+            '200': {
+              description: 'Array of Category model instances',
+              content: {
+                'application/json': {
+                  schema: {
+                    $ref: '#/definitions/Todo',
+                    definitions: {
+                      Todo: {
+                        title: 'Todo',
+                        properties: {
+                          title: {type: 'string'},
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        })
+        async find(): Promise<object[]> {
+          return []; // dummy implementation, it's never called
+        }
+
+        @get('/todos/{id}', {
+          responses: {
+            '200': {
+              content: {
+                'application/json': {
+                  schema: {$ref: '#/components/schemas/Todo'},
+                },
+              },
+            },
+          },
+        })
+        async findById(): Promise<Todo> {
+          return new Todo();
```

Comment:
We can either not label this as an `async` function or do `return await Promise.resolve(new Todo());`

Reply:
The current implementation is fine. `async` has two effects:

1. Allows `await`
2. Forces the return type to a Promise.

Reply:
Oh did not know that, thanks for the clarification :-).